src/main/java/com/github/manolo8/darkbot/extensions/plugins/PluginUpdater.java

```diff
@@ -0,0 +1,258 @@
+package com.github.manolo8.darkbot.extensions.plugins;
+
+import com.github.manolo8.darkbot.Main;
+import com.github.manolo8.darkbot.gui.plugins.PluginCard;
+import com.github.manolo8.darkbot.gui.plugins.PluginDisplay;
+import com.github.manolo8.darkbot.utils.FileUtils;
+import com.github.manolo8.darkbot.utils.I18n;
+import com.github.manolo8.darkbot.utils.Time;
+
+import javax.swing.*;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+
+//todoo i18n
+public class PluginUpdater {
+
+    public final List<PluginException> UPDATING_EXCEPTIONS = new ArrayList<>();
+    private long lastChecked = System.currentTimeMillis();
+
+    private final PluginHandler pluginHandler;
+    private PluginDisplay pluginDisplay;
+
+    private JProgressBar mainProgressBar;
+
+    public PluginUpdater(Main main) {
+        this.pluginHandler = main.pluginHandler;
+    }
+
+    public void scheduleUpdateChecker() {
+        Timer timer = new Timer();
+        timer.schedule(new TimerTask() {
+            @Override
+            public void run() {
+                checkUpdates();
+            }
+        }, 0, Time.DAY);
+    }
+
+    public void setup(PluginDisplay display) {
+        pluginDisplay = display;
+        mainProgressBar = display.getMainProgressBar();
+    }
+
+    public boolean hasAnyUpdates() {
+        return pluginHandler.LOADED_PLUGINS.stream()
+                .anyMatch(pl -> pl.getUpdateStatus() == Plugin.UpdateStatus.AVAILABLE ||
+                        pl.getUpdateStatus() == Plugin.UpdateStatus.INCOMPATIBLE ||
+                        pl.getUpdateStatus() == Plugin.UpdateStatus.FAILED);
+    }
+
+    public long getLastChecked() {
+        return lastChecked;
+    }
+
+    public void checkUpdates() {
+        UPDATING_EXCEPTIONS.clear();
+        new SwingWorker<Void, Void>() {
+            @Override
+            protected void done() {
+                pluginDisplay.refreshUI();
+            }
+
+            @Override
+            protected Void doInBackground() {
+                for (Plugin plugin : pluginHandler.LOADED_PLUGINS) {
+                    plugin.getUpdateIssues().getIssues()
+                            .removeIf(pl -> pl.getMessage().equals(I18n.get("plugins.update_issues.download_failed")));
```

Comment:
Make this check against a constant for the plugin issue, doesn't seem like this has parameters